p2p/peer_tracker.go

```diff
@@ -180,12 +188,40 @@ func (p *peerTracker) gc() {
 				}
 			}
 			p.peerLk.Unlock()
+
+			p.dumpPeers()
 		}
 	}
 }
 
+// dumpPeers stores peers to the PeerTracker's PeerIDStore if
+// it exists.
+func (p *PeerTracker) dumpPeers() {
+	if p.pidstore == nil {
+		return
+	}
+
+	peers := make([]peer.ID, 0, len(p.trackedPeers))
+
+	p.peerLk.RLock()
+	for id := range p.trackedPeers {
+		peers = append(peers, id)
+	}
+	p.peerLk.RUnlock()
+
+	ctx, cancel := context.WithTimeout(p.ctx, time.Second*5)
+	defer cancel()
+
+	err := p.pidstore.Put(ctx, peers)
+	if err != nil {
+		log.Errorw("failed to store tracked peers to PeerIDStore", "err", err)
+		return
+	}
+	log.Debugw("stored peers to PeerIDStore", "amount", len(peers))
+}
+
 // stop waits until all background routines will be finished.
-func (p *peerTracker) stop(ctx context.Context) error {
+func (p *PeerTracker) stop(ctx context.Context) error {
```

Comment:
We should dump on stop as well to guarantee the state is final. The primary purpose of the periodic dump is to tolerate ungraceful shutdowns, but depending on it only is not enough for 5 mins cycle.

Reply:
done - but i did not reduce cycle. I think every 1 min is overkill

Reply:
I don't mind 5 min